packages/realm/src/Logger.ts

```diff
@@ -40,7 +167,60 @@ export enum NumericLogLevel {
  */
 export type Logger = (level: NumericLogLevel, message: string) => void;
 
-export type LoggerCallback = (level: LogLevel, message: string) => void;
+/**
+ * A callback passed to `Realm.setLogger`.
+ * @param level   - The level of the log entry.
+ * @param message - The message of the log entry.
+ * @since 12.0.0
+ * @deprecated Will be removed in v13.0.0
+ */
+export type LoggerCallback1 = (level: LogLevel, message: string) => void;
+
+/**
+ * Represents an entry in the log.
+ */
+export type LogEntry = {
+  /**
+   * The category (origin) of the log entry.
+   */
+  category: LogCategory;
+  /**
+   * The level of the log entry.
+   */
+  level: LogLevel;
+  /**
+   * The message of the log entry.
+   */
+  message: string;
+};
+
+/**
+ * A callback passed to `Realm.setLogger`. Arguments are passed as an object.
```

Comment:
I think the argument itself can be explained by the type instead 🙂 
```suggestion
 * A callback to be used as the logger.
```